src/MCPClient/lib/clientScripts/sanitizeObjectNames.py

```diff
@@ -40,59 +38,68 @@
-    relativeReplacement = objectsDirectory.replace(sipPath, groupType, 1) #"%SIPDirectory%objects/"
+    objectsDirectory = sys.argv[1]  # directory to run sanitization on.
+    sipUUID = sys.argv[2]  # %SIPUUID%
+    date = sys.argv[3]  # %date%
+    taskUUID = sys.argv[4]  # %taskUUID%, unused
+    groupType = sys.argv[5]  # SIPDirectory or transferDirectory
+    groupType = "%%%s%%" % (groupType)  # %SIPDirectory% or %transferDirectory%
+    groupSQL = sys.argv[6]  # transfer_id or sip_id
+    sipPath = sys.argv[7]  # %SIPDirectory%
 
+    relativeReplacement = objectsDirectory.replace(sipPath, groupType, 1)  # "%SIPDirectory%objects/"
 
     sanitizations = sanitizeNames.sanitizeRecursively(objectsDirectory)
+    for oldfile, newfile in sanitizations.items():
+        logger.info('sanitizations: %s -> %s', oldfile, newfile)
 
-    eventDetail= "program=\"sanitizeNames\"; version=\"" + sanitizeNames.VERSION + "\""
-    for oldfile, newfile in sanitizations:
-        if os.path.isfile(newfile):
-            oldfile = oldfile.replace(objectsDirectory, relativeReplacement, 1)
-            newfile = newfile.replace(objectsDirectory, relativeReplacement, 1)
-            print(oldfile, " -> ", newfile)
+    eventDetail = 'program="sanitizeNames"; version="' + sanitizeNames.VERSION + '"'
 
-            if groupType == "%SIPDirectory%":
-                updateFileLocation(oldfile, newfile, "name cleanup", date, "prohibited characters removed:" + eventDetail, fileUUID=None, sipUUID=sipUUID)
-            elif groupType == "%transferDirectory%":
-                updateFileLocation(oldfile, newfile, "name cleanup", date, "prohibited characters removed:" + eventDetail, fileUUID=None, transferUUID=sipUUID)
-            else:
-                print("bad group type", groupType, file=sys.stderr)
-                exit(3)
+    kwargs = {
+        groupSQL: sipUUID,
+        "removedtime__isnull": True,
+    }
+    for f in File.objects.filter(**kwargs):
+        # Check all files to see if any parent directory had a sanitization event
+        current_location = f.currentlocation.replace(groupType, sipPath)
+        sanitized_location = unicodeToStr(current_location)
+        logger.info('Checking %s', current_location)
 
-        elif os.path.isdir(newfile):
-            oldfile = oldfile.replace(objectsDirectory, relativeReplacement, 1) + "/"
-            newfile = newfile.replace(objectsDirectory, relativeReplacement, 1) + "/"
-            directoryContents = []
+        # Check parent directories
+        # Since directory keys are a mix of sanitized and unsanitized, this is a little complicated
+        # Directories keys are in the form sanitized/sanitized/unsanitized
+        # When a match is found (eg 'unsanitized' -> 'sanitized') reset the search
+        # This will find 'sanitized/unsanitized2' -> 'sanitized/sanitized2' on the next pass
+        # TODO This should be checked for a more efficient solution
```

Comment:
Since every time we find a match we restart the search, I think this might be n^2 on the length of the path.